guides/subscriptions/multi_tenant.md

```diff
@@ -0,0 +1,119 @@
+---
+layout: guide
+doc_stub: false
+search: true
+section: Subscriptions
+title: Multi-Tenant
+desc: Switching tenants in GraphQL Subscription execution
+index: 8
+---
+
+In a multi-tenant system, data from many different accounts is stored on the same server. (An account might be an organization, a customer, a namespace, a domain, etc -- these are all _tenants_.) Gems like [Apartment](https://github.com/influitive/apartment) assist with this arrangement, but it can also be implemented in the application. Here are a few considerations for this architecture when using GraphQL subscriptions.
+
+#### Tenant-based `subscription_scope`
+
+When subscriptions are delivered, {% internal_link "`subscription_scope`",  "subscriptions/subscription_classes#scope" %} is one element used to route data to the right subscriber. In short, it's the _implicit_ identifier for the receiver. In a multi-tenant architecture, `subscription_scope` should reference the context key that names the tenant, for example:
+
+```ruby
+class BudgetWasApproved < GraphQL::Schema::Subscription
+  subscription_scope :tenant # This would work with `context[:tenant] => "acme-corp"`
+  # ...
+end
+
+# Include the scope when `.trigger`ing:
+BudgetSchema.subscriptions.trigger(:budget_was_approved, {}, { ... }, scope: "acme-corp")
+```
+
+
+Alternatively, `subscription_scope` might name something that _belongs_ to the tenant:
+
+```ruby
+class BudgetWasApproved < GraphQL::Schema::Subcription
+  subscription_scope :project_id # This would work with `context[:project_id] = 1234`
+end
+
+# Include the scope when `.trigger`ing:
+BudgetSchema.subscriptions.trigger(:budget_was_approved, {}, { ... }, scope: 1234)
+```
+
+As long as `project_id` is unique among _all_ tenants, that would work fine too. But _some_ scope is required so that subscriptions can be disambiguated between tenants.
+
+#### Choosing a tenant for execution
+
+There are a few places where subscriptions might need to load data:
+
+- When building the payload for the subscription (fetching data to prepare the result)
+- `ActionCableSubscriptions`: when deserializing the JSON string broadcasted by `ActionCable`
+- `PusherSubscriptions` and `AblySubscriptions`: when deserializing query context
+
+Each of these operations will need to select the right tenant in order to load data properly.
+
+For __building the payload__, use a {% internal_link "Tracer", "queries/tracing" %}:
+
+```ruby
+class TenantSelectionTracer
```

Comment:
Probably worth mentioning that this will require `tenant` adding into the `context` build in `GraphqlController` too.